src/types/keyValue/keyValue.middleware.js

```diff
@@ -63,16 +63,40 @@ const reducer = key => prefix => name => defaultData =>
   (state = defaultState(key, defaultData), { type, payload } = {}) => {
```

Comment:
Right now I have the feeling we can do an other function.

- **main**: default behaviour and calls the second fonction base on the parameter type
- **reduceWithArray**: for array type
- **reduceWithSingleIntance**: for single instance parameter

I have the feeling the code would be prettier to read

Reply:
Or factorizing this :
```js
(newState, entity) =>
            keyAlreadyExists(state)(key, entty[key])
              ? update(key, state, entity)
              : add(key, newState, entity)
```

```js
const applyReduce = (state, key, entty) => (existCallback, notExistCallback) =>  (newState, entity) =>
            keyAlreadyExists(state)(key, entty[key])
              ? existCallback(key, state, entity)
              : notExistCallback(key, newState, entity),

// ...
return payload.reduce(applyReduce(state, key, entty)(update, add))

```